R/json-joins.R

```diff
@@ -0,0 +1,40 @@
+#' JSON joins
+#'
+#' These are wrappers around `pack_join()` and `dplyr::nest_join()` which store
+#' the joined data into a json column.
+#' `json_pack_join()` returns all rows and columns in x with a new json columns that contains all packed matches from y.
+#' `json_nest_join()` returns all rows and columns in x with a new json columns that contains all nested matches from y.
```

Comment:
For `dm_wrap_tbl()` and friends I opted for a separate help page for each verb, and also for separate files for each function, even if it becomes slightly redundant. It's easier in the long run.

Reply:
done in https://github.com/cynkra/dm/pull/918/commits/4f51a4098013f885f8a619e05f5fd00673828c7a and https://github.com/cynkra/dm/pull/918/commits/16312a8094bc2f70713d201fe6d9a9dabdef96dc